src/sincosint.jl

```diff
@@ -0,0 +1,256 @@
+                                               0.10197205463267975592E11,
+                                               0.30229865264524075951E12,
+                                               0.27504053804288471142E13,
+                                               0.21818989704686874983E13) /
+                                 @horner(invt, 1.00000000000000000000E0,
+                                               0.11223059690217167788E4,
+                                               0.43685270974851313242E6,
+                                               0.74654702140658116258E8,
+                                               0.58580034751805687471E10,
+                                               0.20157980379272098841E12,
+                                               0.26229141857684496445E13,
+                                               0.87852907334918467516E13)*invt) -
+            cos(x)*invt * (1.0 - @horner(invt, 0.59999999999999993089E1,
+                                               0.96527746044997139158E4,
+                                               0.56077626996568834185E7,
+                                               0.15022667718927317198E10,
+                                               0.19644271064733088465E12,
+                                               0.12191368281163225043E14,
+                                               0.31924389898645609533E15,
+                                               0.25876053010027485934E16,
+                                               0.12754978896268878403E16) /
+                                 @horner(invt, 1.00000000000000000000E0,
+                                               0.16287957674166143196E4,
+                                               0.96636303195787870963E6,
+                                               0.26839734750950667021E9,
+                                               0.37388510548029219241E11,
+                                               0.26028585666152144496E13,
+                                               0.85134283716950697226E14,
+                                               0.11304079361627952930E16,
+                                               0.42519841479489798424E16)*invt)
+    elseif isnan(x)
+        return NaN
+    else
+        return 0.0
+    end
+end
+
+for f in (:sinint, :cosint)
+    @eval begin
+        ($f)(x::Float32) = Float32(($f)(Float64(x)))
+        ($f)(x::Float16) = Float16(($f)(Float64(x)))
+        ($f)(x::Real) = ($f)(float(x))
+        ($f)(x::AbstractFloat) = error("not implemented for ", typeof(x))
+    end
+end
+
+
+"""
+    sinint(x)
+Compute the sine integral function of `x`, defined by ``\\operatorname{Si}(x) := \\int_0^x\\frac{\\sin t}{t} dt``
+for real `x`.
+"""
+sinint
+
+"""
+    cosint(x)
+Compute the cosine integral function of `x`, defined by ``\\operatorname{Ci}(x) := \\gamma + \\log x + \\int_0^x \\frac{\\cos t - 1}{t} dt``
```

Comment:
Yeah, I'd add something like "where ``` ``\\gamma`` ``` is the Euler-Mascheroni constant" to the docstring.